kernel/asm/misc.c

```diff
@@ -31,7 +28,7 @@ pttentry asm_cr3_read(){
 	pttentry cr3;
 	
 	asm volatile(
-		"mov %0, cr3"
+		"mov %%cr3 ,%0"
```

Comment:
Once I merge this, I'm going to add some preprocessor directives so that it still uses Intel syntax on build platforms that support it.  It's just a personal preference on my part, and it also keeps it consistent with the bootloader code.  Hypothetically, clang-darwin might eventually support Intel syntax too, and once it does I think all AT&T inline asm should go for consistency's sake.

Reply:
This is fine.  I wonder if we could externalize all the inline assembler into ASM files.  That way, the differences in how clang and gcc handle inline assembler wouldn't matter